### PR TITLE
Allow analyzer 0.39.x

### PR DESCRIPTION
**Pull request by natebosch**
- Update `build_resolvers` to drop a removed argument. This is the only
  package which now _requires_ the latest analyzer.
- Update all packages with constraints on analyzer to allow `0.39.x`.
- Update packages with a lower bound on analyzer lower than the one from
  `build` so that it has the same lower bound.
- Add dependency overrides so we pick up the latest analyzer to test.

### CODE REVIEW

**Comment by jakemac53** [comment]
It looks like we have to pin some deps in the `example` dir as well for travis

**Comment by natebosch** [comment]
Removing the overrides for `example` and `_test` might be blocked by https://github.com/dart-lang/dart_style/issues/866

File: build_resolvers/lib/src/resolver.dart [hunk 1]
```diff
@@ -260,7 +260,7 @@ Future<void> _createDepsFile(
 List<int> _buildSdkSummary(String dartSdkPath) {
   var resourceProvider = PhysicalResourceProvider.INSTANCE;
   var dartSdkFolder = resourceProvider.getFolder(dartSdkPath);
-  var sdk = FolderBasedDartSdk(resourceProvider, dartSdkFolder, true)
```
[the remark this answers is not in the document]
Comment (natebosch) [reply build_resolvers/lib/src/resolver.dart]: I'm not sure what you were looking for here - is it building a web app that using `dart:html`?

[the remark this answers is not in the document]
Comment (jakemac53) [reply build_resolvers/lib/src/resolver.dart]: Right, like can you correctly resolve types from `dart:html`.

However looking at the analyzer changelog it looks like this is always `true` now so it should be fine.

[the remark this answers is not in the document]
Comment (natebosch) [reply build_resolvers/lib/src/resolver.dart]: yeah, I checked on this change it was safe

**Pull request merged**